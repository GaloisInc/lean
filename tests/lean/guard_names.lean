inductive tree (α : Type)
| leaf : tree
| node (left : tree) (val : α) (right : tree) : tree

constant foo {α : Type} : tree α → tree α

example {α : Type} (a b : tree α) : foo a = a :=
begin
  with_cases { induction a },
  { admit },
  case : l v r ih_l ih_r { trace_state, admit },
end
