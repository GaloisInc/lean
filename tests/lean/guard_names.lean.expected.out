α : Type,
b l : tree α,
v : α,
r : tree α,
ih_l : foo l = l,
ih_r : foo r = r
⊢ foo (tree.node l v r) = tree.node l v r
guard_names.lean:7:0: warning: declaration '[anonymous]' uses sorry
